Code/Core/Process/SystemMutex.cpp

```diff
@@ -58,15 +58,16 @@ bool SystemMutex::TryLock()
 		return true;
     #elif defined( __LINUX__ ) || defined( __APPLE__ )
         AStackString<> tempFileName;
-        tempFileName.Format( "/var/run/%s.lock", m_Name.Get() );
+        tempFileName.Format( "/tmp/%s.lock", m_Name.Get());
         m_Handle = open( tempFileName.Get(), O_CREAT | O_RDWR, 0666 );
         int rc = flock( m_Handle, LOCK_EX | LOCK_NB );
         if ( rc )
         {
-            if ( errno == EWOULDBLOCK )
+            if ( errno == EAGAIN )
```

Comment:
You're rigth, I misread some documentation. The good way seems to be checking both :
From http://www.gnu.org/savannah-checkouts/gnu/libc/manual/html_node/Error-Codes.html : 
Portability Note: In many older Unix systems, this condition was indicated by EWOULDBLOCK, which was a distinct error code different from EAGAIN. To make your program portable, you should check for both codes and treat them the same.